u19_pipeline/puffs.py

```diff
@@ -0,0 +1,82 @@
+    stderr = NULL               : blob                         # stderr for the GUI during the session
+    sync = NULL                 : blob                         # At the start of the session, the software runs multiple python processes. This column contains the times of these processes in seconds 
+    """
+
+    class Trial(dj.Part):
+    	definition = """
+        -> PuffsSession
+        trial_idx            : int                          # trial index, keep the original number in the file
+        ---
+		-> task.TaskLevelParameterSet.proj(level="level")   # The difficulty level of the trial
+        trial_type           : enum('L','R')                # answer of this trial, left or right
+        choice               : enum('L','R','nil')          # choice of this trial, left or right
+        trial_prior_p_left   : float                        # prior probablity of this trial for left
+	    trial_rel_start      : float                        # start time of the trial relative to the beginning of the session [seconds]
+	    trial_rel_finish     : float                        # end time of the trial relative to the beginning of the session [seconds]
+	    trial_duration       : float                        # duration of the trial [seconds]
+	    cue_period           : float                        # duration of cue period [seconds]
+        num_puffs_intended_l : tinyint                      # number of puffs intended on the left side
+	    num_puffs_received_r : tinyint                      # number of puffs actually received on the right side
+	    num_puffs_intended_r : tinyint                      # number of puffs intended on the right side
+	    num_puffs_received_l : tinyint                      # number of puffs actually received on the left side
+	    reward_rel_start     : float                        # timing of reward relative to the beginning of the session [seconds]
+	    reward_scale         : float                        # subject is given 4 microliters * reward_scale as a reward 
+	    rule                 : tinyint                      # 
+        """
+    
+    class TrialPhase(dj.Part):
+    	definition = """
+        -> master.Trial
+        phase                : tinyint                      # phase index, 0=intro, 1=stimulus period, ... see Puffs Task Documentation
+        ---
+	    phase_rel_start      : float                        # start time of the phase relative to the beginning of the session [seconds]
+	    phase_rel_finish     : float                        # end time of the phase relative to the beginning of the session [seconds]
+        """
+    
+    class Puff(dj.Part):
+    	definition = """
+        -> master.Trial
+        puff_idx             : tinyint                     # the index of the puff in this particular trial
+        side                 : tinyint                     # 0 = left side, 1 = right side
```

Comment:
does side have to be in the primary key? `puff_idx` is independently indexed on each side?

Reply:
Good point. Side does not have to be a primary key.